app/models/bank_transaction.rb

```diff
@@ -33,4 +33,55 @@ class BankTransaction < ActiveRecord::Base
   def image_url
     'data:image/png;base64,' + Base64.encode64(self.image)
   end
+
+  def assign_to_invoice
+    return false unless supplier
+
+    content = text
+    content += "\n" + reference if reference.present?
+    invoices = supplier.invoices.unpaid.select {|i| content.include? i.number}
```

Comment:
what do you have in mind exactly?

Reply:
Perhaps a check that the number of characters matched is at least four (and doesn't include spaces). Or add a validation on the invoice model (I don't know if e.g. invoice number `8` would be something appearing in practice). Do you have any experience on that?

Reply:
we have single character numbers in our foodcoop. I personally don't see a problem in the current more greedy way, since we check the amount anyway and if there's so clear way we ignore it.

Reply:
Ok then!